MoreLinq/Traverse.cs

```diff
@@ -0,0 +1,92 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Felipe Sateler. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Traverses a tree starting at the <paramref name="root"/> according to the <paramref name="childrenSelector"/> traversal function.
```

Comment:
Summaries should avoid referencing parameters via `<paramref>` elements because in a [table of summaries](https://morelinq.github.io/1.x/ref/api/html/T_MoreLinq_MoreEnumerable.htm), a reader may not have enough context so using plain language is best.

Consider instead:

> Traverses a tree in breadth-first fashion, starting at a root node and using a user-defined function to get the children at each node of the tree.